FirebaseVertexAI/Sample/ChatSample/Views/ErrorDetailsView.swift

```diff
@@ -0,0 +1,247 @@
+      Text(title)
+        .font(.subheadline)
+      Markdown(value)
+    }
+  }
+}
+
+private struct SafetyRatingsSection: View {
+  var ratings: [SafetyRating]
+
+  var body: some View {
+    Section("Safety ratings") {
+      List(ratings, id: \.self) { rating in
+        HStack {
+          Text("\(String(describing: rating.category))")
+            .font(.subheadline)
+          Spacer()
+          Text("\(String(describing: rating.probability))")
+        }
+      }
+    }
+  }
+}
+
+struct ErrorDetailsView: View {
+  var error: Error
+
+  var body: some View {
+    NavigationView {
+      Form {
+        switch error {
+        case let GenerateContentError.internalError(underlying: underlyingError):
+          Section("Error Type") {
+            Text("Internal error")
+          }
+
+          Section("Details") {
+            SubtitleFormRow(title: "Error description",
+                            value: underlyingError.localizedDescription)
+          }
+
+        case let GenerateContentError.promptBlocked(response: generateContentResponse):
+          Section("Error Type") {
+            Text("Your prompt was blocked")
+          }
+
+          Section("Details") {
+            if let reason = generateContentResponse.promptFeedback?.blockReason {
+              SubtitleFormRow(title: "Reason for blocking", value: reason.rawValue)
+            }
+
+            if let text = generateContentResponse.text {
+              SubtitleMarkdownFormRow(title: "Last chunk for the response", value: text)
+            }
+          }
+
+          if let ratings = generateContentResponse.candidates.first?.safetyRatings {
+            SafetyRatingsSection(ratings: ratings)
+          }
+
+        case let GenerateContentError.responseStoppedEarly(
+          reason: finishReason,
+          response: generateContentResponse
+        ):
+
+          Section("Error Type") {
+            Text("Response stopped early")
+          }
+
+          Section("Details") {
+            SubtitleFormRow(title: "Reason for finishing early", value: finishReason.rawValue)
+
+            if let text = generateContentResponse.text {
+              SubtitleMarkdownFormRow(title: "Last chunk for the response", value: text)
+            }
+          }
+
+          if let ratings = generateContentResponse.candidates.first?.safetyRatings {
+            SafetyRatingsSection(ratings: ratings)
+          }
+
+        case GenerateContentError.invalidAPIKey:
+          Section("Error Type") {
+            Text("Invalid API Key")
+          }
+
+          Section("Details") {
+            SubtitleFormRow(title: "Error description", value: error.localizedDescription)
+            SubtitleMarkdownFormRow(
+              title: "Help",
+              value: """
+              Please provide a valid value for `API_KEY` in the `GenerativeAI-Info.plist` file.
+              """
+            )
+          }
+
+        case GenerateContentError.unsupportedUserLocation:
```

Comment:
I wonder if this is still going to be an issue. Even when providing `europe-west1`, the model happily answers.

Reply:
I removed this case from the switch statement since it'll never be thrown with Vertex AI. Gemini is available on Vertex AI in [these Cloud regions](https://cloud.google.com/vertex-ai/generative-ai/docs/learn/locations#available-regions) but can be called from anywhere (e.g., I could specify `europe-west1` but latency from North America will be higher).